### PR TITLE
add custom root domain

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: _os/os.cockpit/actions.py [hunk 1]
```diff
@@ -53,24 +53,25 @@ def get_dns_clients():
...
         domain = service.hrd.getStr('dns.domain')
+        domain_root = '.' + service.hrd.getStr('dns.root')
+        split_count = domain_root.count('.')
 
-        if not domain.endswith('.barcelona.aydo.com'):  # TODO chagne DNS
-            domain = '%s.barcelona.aydo.com' % domain
-            service.hrd.set('dns.domain', domain)
-        subdomain = '.'.join(domain.split('.', 2)[:2])
+        if not domain.endswith(domain_root):
+            domain = domain + domain_root
```
Comment (rkhamis) [inline _os/os.cockpit/actions.py]: Should be
`domain = '%s.%s' % (domain, domain_root)`

File: _os/os.cockpit/actions.py [hunk 2]
```diff
@@ -53,24 +53,25 @@ def get_dns_clients():
...
+            service.hrd.set('dns.domain', domain)r
```
Comment (rkhamis) [inline _os/os.cockpit/actions.py]: typo?

File: _os/os.cockpit/actions.py (hunk 1 above)
Comment (rkhamis) [inline _os/os.cockpit/actions.py]: No, just noticed it's already prepended

File: _os/os.cockpit/actions.py [hunk 3]
```diff
@@ -53,24 +53,25 @@ def get_dns_clients():
 
         ip = service.parent.hrd.getStr('node.addr')
         domain = service.hrd.getStr('dns.domain')
+        domain_root = '.' + service.hrd.getStr('dns.root')
+        split_count = domain_root.count('.')
```
Comment (rkhamis) [inline _os/os.cockpit/actions.py]: should this not be `domain_root.count('.') - 1` to follow previous logic?

Comment (zaibon) [inline _os/os.cockpit/actions.py]: yep indeed, cause we prepend with . before.
nice catch

File: _os/os.cockpit/actions.py [hunk 4]
```diff
@@ -53,24 +53,25 @@ def get_dns_clients():
...
             service.hrd.set('dns.domain', domain)
-        subdomain = '.'.join(domain.split('.', 2)[:2])
+        subdomain = '.'.join(domain.split('.', split_count)[:split_count])
 
         # set domain to all dns servers
         dns_clients = get_dns_clients()
         for dns_client in dns_clients:
             if domain not in dns_client.domains:
-                domain = dns_client.ensure_domain('aydo.com')
+                domain = dns_client.ensure_domain(domain_root.split('.')[-2:])
```
Comment (rkhamis) [inline _os/os.cockpit/actions.py]: ``.`.join(domain_root.split('.')[-2:])`

**Pull request merged**